Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -627,20 +631,41 @@ open class ComposerVC: _ViewController,
     ///   - typingMention: The potential user mention the current user is typing.
     ///   - mentionRange: The position where the current user is typing a mention to it can be replaced with the suggestion.
     open func showMentionSuggestions(for typingMention: String, mentionRange: NSRange) {
-        userSearchController.search(
-            query: queryForMentionSuggestionsSearch(typingMention: typingMention)
-        )
+        guard let channel = channelController?.channel else {
+            return
+        }
+        guard let currentUserId = channelController?.client.currentUserId else {
+            return
+        }
+        
+        var usersCache: [ChatUser] = []
+
+        if mentionAllAppUsers {
+            userSearchController.search(
+                query: queryForMentionSuggestionsSearch(typingMention: typingMention)
+            )
+        } else {
+            usersCache = searchUsers(
+                channel.watchers.map { $0 } + channel.cachedMembers.map { $0 },
+                by: typingMention,
+                excludingId: currentUserId
+            )
+        }
+
         let dataSource = ChatMessageComposerSuggestionsMentionDataSource(
             collectionView: suggestionsVC.collectionView,
-            searchController: userSearchController
+            searchController: userSearchController,
+            usersCache: usersCache
```

Comment:
Since for now, we are not going to introduce a `MemberSearchController`, for me, it would make sense that this logic leaves inside the `ChatMessageComposerSuggestionsMentionDataSource`. But since this is pretty old code, and IMO it should be refactored, we can leave it like this for now. But it does not make sense that we need to call "search" outside of the data source. Ideally, the data source would have a search() function that would abstract how it is finding the members/users... Sot it would user the `userSearchController` or the `channel.watchers`, so Ideally the data source would have a dependency of `searchController´ and `channel`. Anyways, not important for this PR I guess.